Upgrade sveltekit 2 and svelte 5

sverhoeven opened the pull request:
Refs #31 

TODO
- [x] Upgrade to sveltekit 2, `npx sv migrate sveltekit-2`, https://svelte.dev/docs/kit/migrating-to-sveltekit-2
- [x] Working site
  - [x] The http://localhost:5173/map?nonCO2red=0.5&timing=Delayed page does not render due to Leaflet errors
- [x] Upgrade to svelte 5, `npx sv migrate svelte-5`,  https://svelte.dev/docs/svelte/v5-migration-guide#Migration-script
- [x] Working site
  - [x] new svelte leaflet has internal import error, disabled leafletmap for now
  - [x] infinite redirect loop
  - [x] filter change does not upload line
  - [x] On /regions/USA the display graph checkboxes are showing same line/area.
  - [x] On /map click on effortsharing update url but not data/colors
- [x] Green ci

sverhoeven committed:
Use flask cache to make ws faster (after first call)

On my machine fetch of `/api/USA/effortSharingReductions` went from 115ms to 5ms.

sverhoeven committed:
Only one adapter

sverhoeven committed:
npx sv migrate svelte-5

This migration is experimental — please report any bugs to https://github.com/sveltejs/svelte/issues

Detected SvelteKit 1. You need to upgrade to SvelteKit version 2 first (`npx sv migrate sveltekit-2`).

✔ Run sveltekit-2 migration now? … no
(venv) verhoes@bob:~/git/carbon-budget-explorer/cabe$ npx sv migrate sveltekit-2

This will update files in the current directory
If you're inside a monorepo, run this in individual project directories rather than the workspace root.

Your git working directory is dirty — we recommend committing your changes before running this migration.

✔ Continue? … yes
✔ Which folders should be migrated? › src
Updated @sveltejs/kit to ^2.0.0
Updated @sveltejs/adapter-node to ^2.0.0
Updated vite to ^5.0.0
Updated vitest to ^1.0.0
Added @sveltejs/vite-plugin-svelte version ^3.0.0  (vite-plugin-svelte is a peer dependency of SvelteKit now)
Changed `vitePreprocess` import: https://svelte.dev/docs/kit/migrating-to-sveltekit-2#vitepreprocess-is-no-longer-exported-from-sveltejs-kit-vite
✔ Your project has been migrated

Recommended next steps:

  1: Run npm install (or the corresponding installation command of your package manager)
  2: git commit -m "migration to SvelteKit 2"
  3: Review the migration guide at https://svelte.dev/docs/kit/migrating-to-sveltekit-2
  4: Read the updated docs at https://svelte.dev/docs/kit

Run git diff to review changes.

sverhoeven committed:
Move prerender to ts

sverhoeven committed:
Timing threw missing label errors, removed it

Peter9192 commented:
> due to Leaflet errors

I remember there was a feature request on changing projections. Perhaps we could reconsider using d3 world map instead? If so, we should not spend too much time on fixing leaflet

sverhoeven commented:
Also need to upgrade to latest node LTS version. Somethings now require node 18

sverhoeven committed:
npx sv migrate svelte-5

Need to install the following packages:
sv@0.6.11
Ok to proceed? (y)

added 44 packages, and audited 45 packages in 4s

3 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

This will update files in the current directory
If you're inside a monorepo, don't run this in the root directory, rather run it in all projects independently.

Your git working directory is dirty — we recommend committing your changes before running this migration.

✔ Continue? … yes
✔ Which folders should be migrated? › src, static, tests
✔ Do you want to use the migration tool to convert your Svelte components to the new syntax? (You can also do this per component or sub path later) … yes
Updated svelte to ^5.0.0
Updated svelte-check to ^4.0.0
Updated @sveltejs/kit to ^2.5.27
Updated @sveltejs/vite-plugin-svelte to ^4.0.0
Updated prettier to ^3.1.0
Updated prettier-plugin-svelte to ^3.2.6
Updated eslint-plugin-svelte to ^2.45.1
Updated typescript to ^5.5.0
Updated vite to ^5.4.4
✔ Your project has been migrated

Recommended next steps:

  1: install the updated dependencies ('npm i' / 'pnpm i' / etc) (note that there may be peer dependency issues when not all your libraries officially support Svelte 5 yet. In this case try installing with the --force option)
  2: git commit -m "migration to Svelte 5"
  3: Review the migration guide at https://svelte.dev/docs/svelte/v5-migration-guide

Run git diff to review changes.

sverhoeven committed:
Copy configs from freshly made app + fix deps

sverhoeven committed:
Solve some eslint errors/warnings

sverhoeven committed:
Stuck: leaflet kaput + infinite redirect loop + filter change does not update line

sverhoeven committed:
Make flask cachek search string aware

sverhoeven committed:
Make flask cachek search string aware

sverhoeven committed:
Remove log

sverhoeven committed:
Make npm run build succeed, by disabling prerender

Build fails due to page.url used in src/routes/+layout.svelte

sverhoeven committed:
Reload when query changes

sverhoeven committed:
Dont use var before init

sverhoeven committed:
Drop "/<ISO>/<principle>" api endpoint

Was conflicting with caching of "/<ISO>/effortSharings"

sverhoeven committed:
Drop "/<ISO>/<principle>" api endpoint

Was conflicting with caching of "/<ISO>/effortSharings"

sverhoeven committed:
Make leaflet component only render on client, not on server

sverhoeven committed:
Get rid of Error: <path> attribute d: Expected number, "….55454545454545,undefinedL107.92…".

sverhoeven committed:
Simplify changing effort sharing

sverhoeven committed:
Re-enable leaflet component - renders, but no event handling

sverhoeven committed:
Remove orphaned '<'

sverhoeven committed:
Replace svelte-leafletjs with sveaflet, map is back

sverhoeven committed:
Merge branch 'working-dev' into svelte5

sverhoeven committed:
Wrong lang comment char

sverhoeven committed:
When metrics or scale changes then update geojson style

sverhoeven committed:
Make npm run check happy

sverhoeven committed:
format

sverhoeven committed:
npm run lint

Ignore any for event types, can be done later if needed

sverhoeven committed:
Cache forever

sverhoeven committed:
Merge branch 'staging_CW_2_SV' into flask-cache

sverhoeven committed:
Merge remote-tracking branch 'mine/staging_CW_2_SV' into svelte5

sverhoeven committed:
Merge remote-tracking branch 'mine/flask-cache' into svelte5

sverhoeven committed:
Remove flask cache

sverhoeven committed:
Merge remote-tracking branch 'own/svelte5' into svelte5

sverhoeven committed:
Merge remote-tracking branch 'origin/staging' into svelte5

sverhoeven committed:
Merge remote-tracking branch 'origin/staging' into svelte5

sverhoeven committed:
Format

sverhoeven committed:
Merge remote-tracking branch 'origin/staging' into svelte5

sverhoeven committed:
Resolve dep conflict

By syncing with fresh `npx sv create sveltekit2-cabe`

sverhoeven committed:
Make leaflet attribution flagless

Fixes #48

sverhoeven committed:
Remove debug log

sverhoeven committed:
Remove debug log

sverhoeven committed:
Use node 22 for ci

sverhoeven committed:
Merge branch 'svelte5' into flagless-leaflet-attribution

sverhoeven committed:
Use robinson projection

Fixes #49

sverhoeven committed:
Format

sverhoeven committed:
Format

sverhoeven committed:
Require nodejs >=22

sverhoeven committed:
Put Australia on initial map view, loose some vs, canada and mexico

Now new zealand and oceania countries are not visible

sverhoeven committed:
Remove tile layer

As it was showing "artic ocean" label near angola, which means the geojson and tile are using different projections.
To see which country you must hover over it and the name is displayed in upper left corner of map.

sverhoeven committed:
Remove unused import

sverhoeven committed:
Move loading of geojson file to ws.py

Refs #27
Refs #38

sverhoeven committed:
Mention e2e test command

sverhoeven committed:
e2e tests can run without any data

Because e2e test do not test pages that read from ws.py

sverhoeven committed:
Add .env.example

sverhoeven committed:
Use CABE_API_URL env var

Tested
1. by `CABE_API_URL=http://127.0.0.1:5001` in .env
1. running `flask --app ws:app run -p 5001`
1. App start
  1. npm run dev
  2. npm run build && npm run preview
  3. npm run build && node --env-file=.env build/index.js

Visiting global, map, and region page with changing sliders.
Some actions are done server side, some client side via /api proxy.

sverhoeven committed:
Describe the data dir and config

sverhoeven committed:
Read .env and adjust paths

Also
- do not read region files in 2 place
- read aggregated files once
- added convergence year to ECPC of regions effort sharing

sverhoeven committed:
Make `npm run download:borders` download in current working directory

sverhoeven committed:
Apply caching of web service requests in js

sverhoeven committed:
Only read files ending with _GHG_incl.nc

sverhoeven committed:
Make suffix _GHG_incl.nc part of data requirements

sverhoeven committed:
Only cache ok responses

sverhoeven committed:
Make geojson download not happen in slider change

Peter9192 commented on package.json:
```diff
@@ -6,58 +6,60 @@
 		"dev": "vite dev",
 		"build": "vite build",
 		"preview": "vite preview",
+		"prepare": "svelte-kit sync || echo ''",
```
What's that?

sverhoeven replied:
Its part when you run `npx sv create my-app`. I think it generates some Typescript file from the src/routes dir.

Peter9192 commented on src/lib/GlobalBudgetCard.svelte:
```diff
@@ -1,15 +1,25 @@
 <script lang="ts">
+	import { run } from 'svelte/legacy';
```
Why from legacy? Isn't this similar to `$effect`?

Peter9192 commented on src/lib/MiniPathwayCard.svelte:
```diff
@@ -1,30 +1,37 @@
 <script lang="ts">
+	import { run } from 'svelte/legacy';
```
Same, I think it'd be better to do this right, right away: https://svelte.dev/docs/svelte/v5-migration-guide#Migration-script-run

Peter9192 commented on src/lib/PrincipleStatsTable.svelte:
```diff
@@ -1,44 +1,50 @@
 <script lang="ts">
+	import { run } from 'svelte/legacy';
```
Same

Peter9192 commented on src/lib/charts/components/AxisX.svelte:
```diff
@@ -6,40 +6,41 @@
-		? ticks($xScale.ticks())
-		: $xScale.ticks(ticks);
+	/**
+	 * @typedef {Object} Props
```
Why use jsdoc type annotations rather than typescript here?

Peter9192 replied:
Copied from layercake

Peter9192 commented on src/lib/charts/components/NdcRange.svelte:
```diff
@@ -3,29 +3,49 @@
   Generates an SVG area shape using the `area` function from [d3-shape](https://github.com/d3/d3-shape).
  -->
 <script lang="ts">
+	import { handlers } from 'svelte/legacy';
```
another legacy import

Peter9192 commented on src/lib/charts/components/NdcRange.svelte:
```diff
@@ -36,11 +56,13 @@
+		onmouseover={hover}
+		onfocus={() => mouseover()}
+		onmouseout={handlers(
+			() => mouseout(),
```
Why do we have double mouseouts?

Peter9192 commented on src/lib/charts/components/Pie.svelte:
```diff
@@ -29,10 +31,14 @@
-
-	$: slices = pie($animatedData);
-	$: arcGenerator = d3arc<unknown, PieArcDatum<PieData>>().innerRadius(0).outerRadius(radius);
+	run(() => {
```
Use $effect instead?

Peter9192 commented on src/routes/global/+page.svelte:
```diff
@@ -110,24 +118,30 @@
 	const tweenOptions = { duration: 1000, easing: cubicOut };
 	const pathwayCarbonTweened = tweened(data.result.pathwayCarbon, tweenOptions);
-	$: pathwayCarbonTweened.set(data.result.pathwayCarbon);
+	run(() => {
```
use `$effect` or `$derived`?

Peter9192 reviewed (commented):
Runes are more expressive, I'm starting to like that. 

There's a few imports from `legacy`, I think we could replace most instances of `run` with `$effect`. Country page doesn't work for me as I probably have a different data version atm, and not downloading it in the train...

Peter9192 commented on ws.py:
```diff
@@ -16,13 +16,22 @@
+    'CACHE_DIR': './cache'
+})
+cache.init_app(app)
+
```
Caching PR #34 was closed/abandoned but the changes are also present in this branch. Do we still want this in?

sverhoeven committed:
Merge pull request #59 from pbl-nl/flagless-leaflet-attribution

Flagless leaflet attribution

sverhoeven committed:
Merge remote-tracking branch 'origin/svelte5' into robinson-projection

sverhoeven committed:
Merge pull request #60 from pbl-nl/robinson-projection

Use robinson projection

sverhoeven committed:
Merge remote-tracking branch 'origin/svelte5' into datadir-restructure

sverhoeven committed:
Merge branch 'datadir-restructure' into bff-caching

sverhoeven committed:
Add back iso2, used for https://flagcdn.com/${iso2}.svg

sverhoeven committed:
Update README.md

Co-authored-by: Peter Kalverla <peter@kalversweep.nl>

sverhoeven committed:
Merge pull request #62 from pbl-nl/bff-caching

Caching in backend for frontend

sverhoeven committed:
Merge pull request #63 from pbl-nl/borders-once2

Make geojson download not happen on filter change

sverhoeven committed:
Hover is back and hover on map highlights border

sverhoeven committed:
Merge remote-tracking branch 'refs/remotes/origin/datadir-restructure' into datadir-restructure

Peter9192 committed:
Encode new data structure in README

Peter9192 committed:
Update example .env file

Peter9192 committed:
Update web service to use new data directory structure

Peter9192 committed:
Merge pull request #61 from pbl-nl/datadir-restructure

Datadir restructure and configurable via .env
